lib/consolidate.js

```diff
@@ -91,13 +93,24 @@ function read(path, options, fn) {
   if (cached) return fn(null, str);
 
   // read
-  fs.readFile(path, 'utf8', function(err, str){
-    if (err) return fn(err);
-    // remove extraneous utf8 BOM marker
-    str = str.replace(/^\uFEFF/, '');
-    if (options.cache) readCache[path] = str;
-    fn(null, str);
-  });
+  if (nowReading[path]) {
+    if (!readListeners[path]) readListeners[path] = [fn];
+    else readListeners[path].push(fn);
+  }else{
+    nowReading[path] = true;
+    fs.readFile(path, 'utf8', function(err, str){
+      if (err) return fn(err);
```

Comment:
I might be missing something, but shouldn't the error handling be checking an error type and if it's something like described in the subject (`'EMFILE'`) then push it onto `nowReading[path]`, so it can be tried again?
Is there a way to test this situation? I'm trying to understand when this will happen.